plugins/browser-plugin-media-tracking/src/helperFunctions.ts

```diff
@@ -0,0 +1,102 @@
+import { DefaultEvents, EventGroups } from './eventGroups';
+import { DocumentEvent, MediaEvent, TextTrackEvent } from './mediaEvents';
+import { SnowplowMediaEvent } from './snowplowEvents';
+import { EventGroup, RecievedTrackingOptions, TextTrackObject, TrackingOptions } from './types';
+
+export function timeRangesToObjectArray(t: TimeRanges): { start: number; end: number }[] {
+  let out = [];
+  for (let i = 0; i < t.length; i++) {
+    out.push({ start: t.start(i), end: t.end(i) });
+  }
+  return out;
+}
+
+export function textTrackListToJson(textTrackList: TextTrackList): TextTrackObject[] {
+  let out: TextTrackObject[] = [];
+  for (let o of Object.keys(textTrackList)) {
+    let i = parseInt(o);
+    out.push({
+      label: textTrackList[i].label,
+      language: textTrackList[i].language,
+      kind: textTrackList[i].kind,
+      mode: textTrackList[i].mode,
+    });
+  }
+  return out;
+}
+
+export function isTypeTextTrackEvent(e: string): boolean {
+  let fields: string[] = Object.keys(TextTrackEvent);
+  return fields.indexOf(e) !== -1;
+}
+
+export function isTypeDocumentEvent(e: string): boolean {
+  let fields: string[] = Object.keys(DocumentEvent);
+  return fields.indexOf(e) !== -1;
+}
+
+function elementsInArrayOutOfBounds(arr: number[]): boolean {
+  return arr.filter((a) => a < 1 || 100 < a).length !== 0;
+}
+
+function duplicatesInArray(arr: number[]): boolean {
+  return arr.filter((b) => arr.filter((f) => f == b).length > 1).length !== 0;
+}
+
+export function isElementFullScreen(mediaId: string): boolean {
+  if (document.fullscreenElement) {
+    return document.fullscreenElement.id === mediaId;
+  }
+  return false;
+}
+
+export function boundryErrorHandling(boundries: number[]): number[] {
+  if (duplicatesInArray(boundries)) {
+    boundries = boundries.filter((item, pos, self) => self.indexOf(item) == pos);
+    console.error('Duplicate values found in boundry array');
```

Comment:
When I see something like this I usually end up wonder if we actually need this function at all. I know its there for safety but I'm not a huge fan of runtime error checking, it adds bloat and doesn't add much value given usage should be documented well. Unless of course we can recover effectively. So here we do manage to recover nicely, not sure we need the errors in that case, because we're just helping by trimming out invalid numbers which would never make any sense anyway.

I think we can safely remove the console.error messages here (saves a few bytes too!)